dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java

```diff
@@ -15,25 +19,64 @@
-                Logger.info(DotRestHighLevelClientProvider.class,
-                        "RestHighLevelClientProvider " + providerClassName + " loaded successfully");
-            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
-                INSTANCE = new DotRestHighLevelClientProvider();
-                Logger.error(RestHighLevelClientProvider.class,
-                        "Unable to get the class reference for the RestHighLevelClientProvider  ["
-                                + providerClassName + "].", e);
+            synchronized (RestHighLevelClientProvider.class) {
+                if (INSTANCE == null) {
+
+                    final String providerClassName = Config.getStringProperty("ES_REST_CLIENT_PROVIDER_CLASS",
+                                    "com.dotcms.content.elasticsearch.util.DotRestHighLevelClientProvider");
+
+                    disableStringIntern();
+
+                    try {
+                        INSTANCE = ((Class<RestHighLevelClientProvider>) Class.forName(providerClassName)).newInstance();
+                        Logger.info(DotRestHighLevelClientProvider.class,
+                                        "RestHighLevelClientProvider " + providerClassName + " loaded successfully");
+                    } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                        INSTANCE = new DotRestHighLevelClientProvider();
+                        Logger.error(RestHighLevelClientProvider.class,
+                                        "Unable to get the class reference for the RestHighLevelClientProvider  ["
+                                                        + providerClassName + "].",
+                                        e);
+                    }
+                }
             }
         }
-
+        
         return INSTANCE;
     }
 
+    /**
+     * This method uses reflection to disable String.intern in elasticsearch's Jackson JsonFactory
+     * instance. If this method fails for any reason, it is not tragic, just not optimal. Also, this is
+     * hard to test for - the only indication that any change has happened to the JsonFactory is an
+     * inner flag on the JsonFactory object itself
+     * 
+     * @return
+     */
+    private static boolean disableStringIntern() {
+        try {
+            // init the static field
+            XContentBuilder x = JsonXContent.contentBuilder();
+
+            // get a handle on the field
+            Field jsonFactoryField = JsonXContent.class.getDeclaredField("jsonFactory");
+            jsonFactoryField.setAccessible(true);
+            JsonFactory jsonFactory = (JsonFactory) jsonFactoryField.get(null);
+
+            // set the value
+            Logger.info(RestHighLevelClientProvider.class, "disabling String.intern() in Jackson");
+            jsonFactory.configure(JsonFactory.Feature.INTERN_FIELD_NAMES, false);
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4883429)